### PR TITLE
Introduce a method to get a relative URI out of absolute ones

### PR DESCRIPTION
**Pull request by jviotti**
Signed-off-by: Juan Cruz Viotti <jv@jviotti.com>

### CODE REVIEW

File: src/uri/uri.cc [hunk 1]
```diff
@@ -408,6 +408,33 @@ auto URI::resolve_from(const URI &base) -> URI & {
   }
 }
 
+auto URI::relative_to(const URI &base) -> URI & {
+  UriUriA result;
+
+  // According to the docs, we only need to free on success
+  // See
+  // https://uriparser.github.io/doc/api/latest/Uri_8h.html#af948b48ec0749192f1a944b044d5726f
+  switch (uriRemoveBaseUriMmA(&result, &this->internal->uri,
```
[the remark this answers is not in the document]
Comment (jviotti) [reply src/uri/uri.cc]: For sure, which is what you were working towards on the last PR. I think we will be able to get rid of it at some point soon.

File: test/uri/uri_relative_to_test.cc [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  EXPECT_EQ(uri.recompose(), "foo");
+}
+
+TEST(URI_relative_to, relative_absolute_1) {
+  const sourcemeta::jsontoolkit::URI base{"foo/bar"};
+  sourcemeta::jsontoolkit::URI uri{"https://www.example.com/foo"};
+  uri.relative_to(base);
+  EXPECT_EQ(uri.recompose(), "https://www.example.com/foo");
+}
+
```
Comment (tony-go) [inline test/uri/uri_relative_to_test.cc:66]: I think We should have better name than this :)

**Review by tony-go — APPROVED** [review]
One nitpick!

File: test/uri/uri_relative_to_test.cc (hunk 2 above)
Comment (jviotti) [reply test/uri/uri_relative_to_test.cc:66]: What do you mean?

Comment (tony-go) [reply test/uri/uri_relative_to_test.cc:66]: Like explicitly mentioning what we are testing ‘relative_schemeless_uri’ or whatever make sense :)

Comment (jviotti) [reply test/uri/uri_relative_to_test.cc:66]: Ah, yeah, fair enough. We should also do a lot more testing once we implement this purely ourselves. I wonder what edge cases `uriparser` is already sorting out here that we are not aware of
